app/src/main/java/com/mapbox/maps/testapp/examples/terrain/SantaCatalinaActivity.kt

```diff
@@ -0,0 +1,311 @@
+package com.mapbox.maps.testapp.examples.terrain
+
+import android.animation.*
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import android.graphics.Color.rgb
+import android.os.Bundle
+import android.view.Window
+import androidx.annotation.ColorInt
+import androidx.annotation.DrawableRes
+import androidx.appcompat.app.AppCompatActivity
+import com.mapbox.api.directions.v5.DirectionsCriteria
+import com.mapbox.api.directions.v5.MapboxDirections
+import com.mapbox.api.directions.v5.models.DirectionsResponse
+import com.mapbox.core.constants.Constants.PRECISION_6
+import com.mapbox.geojson.LineString
+import com.mapbox.geojson.Point
+import com.mapbox.maps.*
+import com.mapbox.maps.extension.style.image.image
+import com.mapbox.maps.extension.style.layers.generated.*
+import com.mapbox.maps.extension.style.layers.getLayerAs
+import com.mapbox.maps.extension.style.layers.properties.generated.Visibility
+import com.mapbox.maps.extension.style.sources.addSource
+import com.mapbox.maps.extension.style.sources.generated.geoJsonSource
+import com.mapbox.maps.extension.style.sources.generated.rasterDemSource
+import com.mapbox.maps.extension.style.style
+import com.mapbox.maps.extension.style.terrain.generated.terrain
+import com.mapbox.maps.plugin.compass.getCompassPlugin
+import com.mapbox.maps.plugin.gestures.getGesturesPlugin
+import com.mapbox.maps.plugin.location.utils.BitmapUtils
+import com.mapbox.maps.plugin.scalebar.getScaleBarPlugin
+import com.mapbox.maps.testapp.R
+import com.mapbox.turf.TurfConstants
+import com.mapbox.turf.TurfMeasurement
+import kotlinx.android.synthetic.main.activity_simple_map.*
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+/**
+ * Example showcasing terrain with animating the free camera with a line string.
+ */
+class SantaCatalinaActivity : AppCompatActivity() {
+
+  private lateinit var mapboxMap: MapboxMap
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    requestWindowFeature(Window.FEATURE_NO_TITLE)
+    setContentView(R.layout.activity_simple_map)
+    mapView.disablePlugins()
+
+    // get map and setup initial camera
+    mapboxMap = mapView.getMapboxMap()
+    mapboxMap.jumpTo(
+      CameraOptions.Builder()
+        .center(POINT_START)
+        .zoom(14.0)
+        .bearing(215.0)
+        .build()
+    )
+
+    // load satellite style and add terrain with a line layer to visualize the route
+    mapboxMap.loadStyle(
+      style(styleUri = Style.SATELLITE_STREETS) {
+        +rasterDemSource(SOURCE) {
+          url(TERRAIN_URL_TILE_RESOURCE)
+          tileSize(512)
+        }
+        +terrain(SOURCE) {
+          exaggeration(1.7)
```

Comment:
Do we actually need it?
If yes - move to const.

Reply:
need it, no but it looks cooler